hack/providers-sync-tools/README.md

```diff
@@ -0,0 +1,56 @@
+# Providers Sync Tools
+
+## Overview
+
+Tanzu-framework is moving towards packaging cluster-api componentry. While this
+transition is happening, tanzu-framework will have cluster-api components in
+two locations: `packages/management/cluster-api*` and `pkg/v1/providers`.
+
+These tools exist to ensure cluster-api related packages in the
+`packages/management` directory stay in sync with the content in
+`pkg/v1/providers`.
+
+## How it works
+
+The tooling in this directory vendirs the cluster-api packages into a git
+ignored directory:
+
+`hack/provider-sync-tools/<some cluster-api package>/build/upstream`
+
+The tooling then renders the files into `hack/provider-sync-tools/<some
+cluster-api-package>/build/rendered` using ytt. The render includes two sets of
+overlays.
+
+1. The first set of overlays come from the package's overlays.
+
+1. The second set of overlays are specific to edits needed to
+   `pkg/v1/providers`, and exist in the `hack/provider-sync-tools/<some
+   cluster-api package>/overlay`. An example of such an edit would be the
+   controller image in the CRDs of the package.
+
+This tooling will prevent edits made directly to the cluster-api related
+folders in `pkg/v1/providers`. Instead of directly editing those files,
+overlays should be created in `hack/provider-sync-tools/<some cluster-api
+package>/overlay`. Running the `make all` target in this folder will render the
+package contents along with the overlays, and then place the edited files in
+`pkg/v1/providers`.
+
+The validate task will ensure that the rendered content from the package and
+overlays have no diff with the content of `pkg/v1/providers`.
+
+## What to do in common scenarios
+
+1. Making a tweak to `pkg/v1/providers/<cluster-api-folder>`
```

Comment:
nit: might be good to qualify that for infra providers, only tweaks that involves modifying infrastructure-components needs to follow this procedure, since the majority of changes related to a specific infra provider is usually to tweak files involved in the cluster configuration generation, which this tool does not process.

Reply:
Good point, The README.md was updated to convey how the tools should be used.